zipline/utils/sentinel.py

```diff
@@ -7,19 +7,33 @@
 
 
 def sentinel(name, doc=None):
-    @object.__new__   # bind a single instance to the name 'NotSpecified'
-    class result(object):
+    try:
+        value = sentinel._cache[name]  # memoized
+    except KeyError:
+        pass
+    else:
+        if doc == value.__doc__:
+            return value
+
+        raise ValueError(
+            'attempted to create sentinel with a used name and a different'
+            ' docstring: %r\noriginal docstring:\n%s' % (name, value.__doc__),
+        )
+
+    @object.__new__   # bind a single instance to the name 'Sentinel'
+    class Sentinel(object):
```

Comment:
the `__name__` is only on the instance. This means that `type(sentinel('a')).__name__ == 'Sentinel'` so they are a little different. Also, this is giving us an instance directly without invoking the `__new__`. I just kind of prefer the class syntax but I don't have a huge attachment.